pytorch_lightning/trainer/training_io.py

```diff
@@ -401,10 +402,14 @@ def restore_training_state(self, checkpoint):
                 ' This is probably due to `ModelCheckpoint.save_weights_only` being set to `True`.'
             )
 
-        if self.checkpoint_callback is not None and self.checkpoint_callback is not False:
-            self.checkpoint_callback.best = checkpoint['checkpoint_callback_best']
+        if self.checkpoint_callback:
+            if 'checkpoint_callback_best' in checkpoint:
+                self.checkpoint_callback.best_model_score = checkpoint['checkpoint_callback_best']
+            else:
+                self.checkpoint_callback.best_model_score = checkpoint['checkpoint_callback_best_model_score']
+            self.checkpoint_callback.best_model_path = checkpoint['checkpoint_callback_best_model_path']
```

Comment:
```suggestion
            self.checkpoint_callback.best_model_path = checkpoint.get('checkpoint_callback_best_model_path')
```
as the old models does not have it

Reply:
True, good catch.